modules/terraform/workspace.go

```diff
@@ -0,0 +1,36 @@
+package terraform
+
+import (
+	"strings"
+	"testing"
+)
+
+// WorkspaceSelectOrNew runs terraform workspace with the given options and returns workspace name.
+// It tries to select a workspace with the given name, or it creates a new one if it doesn't exist.
+func WorkspaceSelectOrNew(t *testing.T, options *Options, name string) string {
+	out, err := WorkspaceSelectOrNewE(t, options, name)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return out
+}
+
+// WorkspaceSelectOrNewE runs terraform workspace with the given options and returns workspace name.
+// It tries to select a workspace with the given name, or it creates a new one if it doesn't exist.
+func WorkspaceSelectOrNewE(t *testing.T, options *Options, name string) (string, error) {
+	out, err := RunTerraformCommandE(t, options, "workspace", "list")
+	if err != nil {
+		return out, nil
+	}
+
+	if strings.Contains(out, name) {
```

Comment:
This seems error prone. If you have a workspace called `foobar` and you set `name` to `foo`, this will return true even though `foo` doesn't actually exist. I think you need to split the output into individual lines and loop over each line looking for an exact match to `name`. Note that the currently selected workspace may have an asterisk next to it, so it might require some extra cleaning.

Reply:
That's correct, I'll modify it.